round4/team-17.md

```diff
@@ -0,0 +1,123 @@
+# Chrome Extension Idea: TabsOFF
```

Comment:
First, I really like this idea. I, myself, struggle with tab organization, as I often have 50 stackoverflow tabs open when programming. I also think this project is very feasible. Off the bat, I think you could use hooks such as [chrome.tabs.onCreated](https://developer.chrome.com/docs/extensions/reference/api/tabs#event-onCreated) and [chrome.tabs.onActivated](https://developer.chrome.com/docs/extensions/reference/api/tabs#event-onActivated) to keep track of the inactive time a user has been on a tab.

I think it would be helpful to add a user preference that automatically closes a tab after x amount of time of inactivity. I would also look into giving users the freedom to blacklist specific important tabs (or pinned tabs) from being closed from your extension. When designing the dashboard to show the activity of tabs, I think it would be helpful to allow the users to manually delete multiple tabs easily. Also consider adding filtering/sorting for the tabs in the dashboard, so users can view/delete specific categories of tabs (perhaps by website or by duration of inactivity).